lib/Twig/Test/IntegrationTestCase.php

```diff
@@ -17,27 +17,52 @@
  */
 abstract class Twig_Test_IntegrationTestCase extends PHPUnit_Framework_TestCase
 {
-    abstract protected function getExtensions();
+    /**
+     * @return string
+     */
     abstract protected function getFixturesDir();
 
     /**
-     * @dataProvider getTests
+     * @return Twig_ExtensionInterface[]
      */
-    public function testIntegration($file, $message, $condition, $templates, $exception, $outputs)
+    protected function getExtensions()
     {
-        $this->doIntegrationTest($file, $message, $condition, $templates, $exception, $outputs);
+        return array();
+    }
+
+    /**
+     * @return Twig_SimpleFilter[]
+     */
+    protected function getFilters()
+    {
+        return array();
     }
 
     /**
-     * @dataProvider getLegacyTests
-     * @group legacy
+     * @return Twig_SimpleFunction[]
      */
-    public function testLegacyIntegration($file, $message, $condition, $templates, $exception, $outputs)
```

Comment:
I see the point of having legacy test, I'm just wondering if `IntegrationTestCase` is the best place to handle those. 
If the `IntegrationTestCase` is meant as end point for others to reuse as base for fixtures tests  (outside of the Twig codebase) then I think the legacy support can better be done in another test.